### DISCUSSION TITLE
Renovate tag count at 10K causing issues

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#14263] [Suggest an Idea]
Today I've identified that the Renovate Docker image tag count recently reaching and exceeding 10,000 may cause issues. My organization uses a mirror repo for Docker Hub using JFrog Container Registry / Artifactory. If the `last` query parameter with the value of the last found item is added to the `/v2/renovate/renovate/tags/list?n=10000` call, Artifactory returns an HTTP 404 error, which causes Renovate to decide the search for tags failed.

I noticed that if the value of `last` is an integer, like the last found major version, then Artifactory does seem to return a valid response. At the same time, that's contradicting the spec as described on (https://docs.docker.com/registry/spec/api/#listing-image-tags): `GET /v2/<name>/tags/list?n=<n from the request>&last=<last tag value from previous response>`

Are you aware of more issues like this. as well as possible workarounds?

**Comment by viceice** [comment]
🤔 looks like a bug in artifactory.

**Reply by astellingwerf** [reply]
That's right. We've reported the issue and JFrog accepted it as a ticket in their backlog.
A workaround is to have Renovate check the version against Docker Hub directly

**Reply by tonnico** [reply]
Maybe the next one is searching also the backlog ticket: https://www.jfrog.com/jira/browse/RTFACT-27035
